benchmarks/profiler/utils/search_space_autogen.py

```diff
@@ -55,62 +56,84 @@ def auto_generate_search_space(args: argparse.Namespace) -> None:
 
+    # now determine the search space
+    if args.enable_gpu_discovery:
+        if (
+            args.min_num_gpus_per_engine == 0
+            or args.max_num_gpus_per_engine == 0
+            or args.num_gpus_per_node == 0
+        ):
+            if not args.model:
+                # TODO: get model info provided DGD config
+                error_msg = "No model provided, cannot auto-generate GPU search space. Please provide `--model` or GPU info"
+                logger.error(error_msg)
+                raise RuntimeError(error_msg)
+
+            logger.info("Getting GPU info from k8s cluster...")
+            gpu_info = get_gpu_summary()
+            logger.info(
+                f"Cluster has {gpu_info['gpus_per_node']}x{gpu_info['model']} GPUs per node with {gpu_info['vram']} VRAM"
+            )
+
+            # model_info should be set by now (checked above), but mypy needs explicit verification
+            assert (
+                model_info is not None
+            ), "model_info must be set when model is provided"
+
+            vram_mib = int(gpu_info["vram"])  # type: ignore[call-overload]
+            gpus_per_node = int(gpu_info["gpus_per_node"])  # type: ignore[call-overload]
+
+            min_gpu = math.ceil(
+                model_info.model_size / MODEL_GPU_MEM_FRAC_MAX / vram_mib
+            )
+            if not model_info.is_moe:
+                max_gpu = gpus_per_node
+            else:
+                max_gpu = max(min_gpu * MOE_MODEL_MAX_NUM_GPU_FACTOR, gpus_per_node)
+            if min_gpu > max_gpu:
+                error_msg = f"No valid GPU configuration found for model {args.model} on the cluster with {gpu_info['gpus_per_node']}x{gpu_info['model']} GPUs per node"
+                logger.error(error_msg)
+                raise RuntimeError(error_msg)
+
+            logger.info(
+                f"Auto-generated search space for model {args.model} on the cluster with {gpu_info['gpus_per_node']}x{gpu_info['model']} GPUs per node: {min_gpu} to {max_gpu}"
+            )
+            args.min_num_gpus_per_engine = min_gpu
+            args.max_num_gpus_per_engine = max_gpu
+            args.num_gpus_per_node = gpus_per_node  # type: ignore[assignment]
+    else:
+        # use default values for GPUs
+        if args.min_num_gpus_per_engine == 0:
+            logger.info(
+                "GPU discover is disabled and min_num_gpus_per_engine is not specified, setting to 1"
+            )
+            args.min_num_gpus_per_engine = 1
+        if args.max_num_gpus_per_engine == 0:
+            logger.info(
+                "GPU discover is disabled and max_num_gpus_per_engine is not specified, setting to 4"
+            )
+            args.max_num_gpus_per_engine = 4
+        if args.num_gpus_per_node == 0:
+            logger.info(
+                "GPU discover is disabled and num_gpus_per_node is not specified, setting to 8"
+            )
+            args.num_gpus_per_node = 8
```

Comment:
I don't like this hardcoding here, why not set these values as defaults in `argparse`?